widgetssdk/src/test/java/com/glia/widgets/callvisualizer/ActivityWatcherControllerTest.kt

```diff
@@ -223,6 +226,120 @@ internal class ActivityWatcherControllerTest {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer, true)
+        verify(watcher).requestCameraPermission()
+        controller.onCameraPermissionResult(false)
+    }
+
+    @Test
+    fun `onCameraPermissionResult is not granted and fail and is not component activity`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer, false)
+        verify(watcher).openComponentActivity()
+        controller.onCameraPermissionResult(false)
+    }
+
+    @Test
+    fun `onMediaUpgradeAccept destroys support activity when error is null`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer, false)
+        resetMocks()
+        controller.onMediaUpgradeAccept(null)
+        verify(watcher).destroySupportActivity()
+    }
+
+    @Test
+    fun `onMediaUpgradeAccept does not destroy support activity when error occurs`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer, true)
+        resetMocks()
+        controller.onMediaUpgradeAccept(GliaException("message", GliaException.Cause.INTERNAL_ERROR))
+    }
+
+    @Test
+    fun `onMediaUpgradeAccept destroys support activity when error occurs`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer, false)
+        resetMocks()
+        controller.onMediaUpgradeAccept(GliaException("message", GliaException.Cause.INTERNAL_ERROR))
+        verify(watcher).destroySupportActivity()
+    }
+
+    @Test
+    fun `onMediaUpgradeAccept does not destroy support activity when error is null`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer, true)
+        resetMocks()
+        controller.onMediaUpgradeAccept(null)
+    }
+
+    @Test
+    fun `onMediaUpgradeDecline destroys support activity when error is null`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer, false)
+        resetMocks()
+        controller.onMediaUpgradeDecline(null)
+        verify(watcher).destroySupportActivity()
+        verify(watcher).dismissAlertDialog(true)
+    }
+
+    @Test
+    fun `onMediaUpgradeDecline does not destroy support activity when error is null`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer, true)
+        resetMocks()
+        controller.onMediaUpgradeDecline(null)
+        verify(watcher).dismissAlertDialog(true)
+    }
+
+    @Test
+    fun `onMediaUpgradeDecline destroys support activity when error occurs`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer, false)
+        resetMocks()
+        controller.onMediaUpgradeDecline(GliaException("message", GliaException.Cause.INTERNAL_ERROR))
+        verify(watcher).destroySupportActivity()
+    }
+
+    @Test
+    fun `onMediaUpgradeDecline does not destroy support activity when error occurs`() {
+        controller.onMediaUpgradeReceived(mediaUpgradeOffer, true)
+        resetMocks()
+        controller.onMediaUpgradeDecline(GliaException("message", GliaException.Cause.INTERNAL_ERROR))
+    }
```

Comment:
Maybe I'm missing something, but why doesn't this test contain the "verify" check? Besides, I'm not sure I understand how is it different from the previous test.

Reply:
A few tests are incomplete due to Mockito not being able to mock direct variable access. The problem comes from:

`if (mediaUpgradeOffer.video != null && mediaUpgradeOffer.video != MediaDirection.NONE) {
                onPositiveDialogButtonClicked()
            } else {`
            
mediaUpgradeOffer has direct access to variables, no setter and no constructor. So currently we can't mock the value or create a value ourselves. The only way would be to hide the value access behind an extenal file that I might actually do, but in general I am against making code changes due to testing requirements

Reply:
Is this test "working" then? I mean, is it actually testing something? If so, what is the mechanics?

Reply:
In other words, how does it make sure that "does not destroy support activity" is true, could you please explain just to understand?

Reply:
I made some name changes and test functionality changes, but still, some tests test until some blocker and tests that all things that can be tested will be tested and otherwise tests that nothing that is not expected is called